docs/JSON_format.md

```diff
@@ -625,23 +625,25 @@ attributes for these objects are:
   - `actions`: the list of actions (order does not matter) supported by this
   table
   - `next_tables`: maps each action to a next table name. Alternatively, maps
-  special string `__HIT__` and `__MISS__` to a next table name.
+  special string `__HIT__` and `__MISS__` to a next table name. See Note 3 below.
   - `direct_meters`: the name of the associated direct meter array, or null if
   the match table has no associated meter array
   - `default_entry`: an optional JSON item which can force the default entry for
   the table to be set when loading the JSON, without intervention from the
   control plane. It has the following attributes:
-    - `action_id`: the id of the default action
+    - `action_id`: the id of the default action.  This is required if
+	`default_entry` is present.
     - `action_const`: an optional boolean value which is `true` iff the control
     plane is not allowed to change the default action function. Default value is
-    `false`. It can only be set to `true` for `simple` tables.
+    `false`. It can only be set to `true` for `simple` tables.  See Note 2 below.
     - `action_data`: an optional JSON array where each entry is the hexstring
     value for an action argument. The size of the array needs to match the
     number of parameters expected by the action function with id `action_id`.
+	This is required if `default_entry` is present.
```

Comment:
I don't think this is correct.
The request was more about including in Note 2 that:
> p4c always include the `action_data` attribute in the `default_entry` object.

Reply:
If we do add this new restriction, we should update the phrasing above ("an optional JSON array").

Reply:
Updated in commit 9.